src/SocialIndexActionButton.jsx

```diff
@@ -0,0 +1,89 @@
+const { actionName, actionUndoName, item, notifyAccountId } = props;
+
+const [isActive, setIsActive] = useState(null);
+const [isProcessingAction, setIsProcessingAction] = useState(false);
+
+const actions = Social.index(actionName, item);
+const isLoading = !actions;
+const actionsByUsers = {};
+(actions || []).forEach((action) => {
+  if (action.value.type === actionName) {
+    actionsByUsers[action.accountId] = action;
+  } else if (action.value.type === actionUndoName) {
+    delete actionsByUsers[action.accountId];
+  }
+});
+
+if (actionsByUsers[context.accountId] && isActive === null) {
+  setIsActive(true);
+}
+
+if (isActive !== null) {
```

Comment:
Could you explain this block? I don't understand its purpose since actionsByUsers was just built from the index.

Reply:
Sure! The logic inside `if (isActive !== null) { ... }` updates `actionsByUsers` whenever the user clicks the button - which happens after the index data is initially fetched. So we need to update the stale data in order for the count to be updated instantly when evaluating `const count = Object.keys(actionsByUsers).length;`.